substrate/frame/support/procedural/src/pallet/expand/config.rs

```diff
@@ -132,6 +156,7 @@ pub fn expand_config_metadata(def: &Def) -> proc_macro2::TokenStream {
 	});
 
 	quote::quote!(
+		#[allow(deprecated)]
```

Comment:
I guess it no longer use the deprecated associated type, and always use system type no?
```suggestion
```